### ISSUE TITLE
Infinite sequence of `END_OBJECT` tokens returned at end of streaming read

### ISSUE THREAD

**Issue by wanglingsong** [FasterXML/jackson-dataformats-binary#135]
Jackson version: 2.9.4
I'm trying to use streaming JsonParser to parse protobuf binary data according to your example, however the nextToken method keep returning "END_OBJECT" token in the end of parsing. You can reproduce this issue by following code:
```java
        ObjectMapper mapper = new ProtobufMapper();
        String protobuf_str = "message Employee {\n"
                + " required string name = 1;\n"
                + " required int32 age = 2;\n"
                + " repeated string emails = 3;\n"
                + " optional Employee boss = 4;\n"
                + "}\n";
        final ProtobufSchema schema = ProtobufSchemaLoader.std.parse(protobuf_str);

        Employee empl = new Employee();
        empl.age = 30;
        empl.emails = new String[]{"foo@gmail.com"};
        empl.name = "foo";

        byte[] protobufData = mapper.writer(schema)
                .writeValueAsBytes(empl);

        JsonParser jsonParser = new ProtobufFactory().createParser(new ByteArrayInputStream(protobufData));
        jsonParser.setSchema(schema);
        JsonToken token;
        while ((token = jsonParser.nextToken()) != null) {
            System.out.println(token.id());
        }
```
output:
```
1
5
6
5
7
5
3
6
4
2
2
2
2
2
2
2
2
2
2
2
2
2
2
...endless
```

**Comment by cowtowncoder** [comment]
Thank you for reporting this. It does sound like a bug.

One minor request: if it is easy enough, could you verify it still fails with 2.9.5? I assume it does as I do not recall fix to specific problem. But there have been recent fixes to protobuf module.

**Comment by wanglingsong** [comment]
Checked. Still fails with 2.9.5

**Comment by cowtowncoder** [comment]
Ok, I can reproduce the issue. Seems to require non-empty array, for what that is worth, probably as the last value.

**Comment by cowtowncoder** [comment]
Fixed for 2.9.6.
